lib/livebook/hubs.ex

```diff
@@ -54,18 +60,31 @@ defmodule Livebook.Hubs do
   def save_hub(struct) do
     attributes = struct |> Map.from_struct() |> Map.to_list()
     :ok = Storage.insert(@namespace, struct.id, attributes)
+    :ok = connect_hub(struct)
     :ok = broadcast_hubs_change()
+
     struct
   end
 
   @doc false
   def delete_hub(id) do
-    Storage.delete(@namespace, id)
+    if hub_exists?(id) do
+      hub = fetch_hub!(id)
```

Comment:
We can have a race condition here. could we use something like:

```suggestion
    if hub = get_hub(id) do
```

instead?

Or:

```suggestion
    with {:ok, hub} <- fetch_hub(id) do
```